### ISSUE TITLE
huge performance issues migrating from RC1 to RC2

### ISSUE THREAD

**Issue by acateland** [angular/angular#9295]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
![rc2](https://cloud.githubusercontent.com/assets/19344650/16146331/c732ec5c-347d-11e6-8c0d-c340719583ec.png)

**Expected/desired behavior**
![rc1](https://cloud.githubusercontent.com/assets/19344650/16146329/c408239e-347d-11e6-8b36-17d8d60c6b3b.png)

**Reproduction of the problem**
simply updating my package.json with RC2 versions of angular modules.
Using the production mode, i can feel and see in devtools a huge increase of javascript execution time.
2 or 3 time more than on RC1

My test application is a svg documents with two thousand elements, mutating svg property on only a few of them at time. (a dozen)

**What is the expected behavior?**
Keeping same level of performances.

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.2
 
* **Browser:** [all] 
 
* **Language:** [ES6/7]

**Comment by zoechi** [comment]
Any concrete repo case (Plunker) that allows to reproduce?

**Comment by acateland** [comment]
@zoechi working on a simplified test case to expose.

**Comment by pkozlowski-opensource** [comment]
Yeh, we will need a minimal reproduce scenario in a plunker, hardly actionable otherwise... Unless this is #9208 (fix in master but not part of rc.2)

**Comment by CaptainCodeman** [comment]
I had what could be the same issue going from RC1 to RC2 - zone / change detection running constantly. It turned out to be a pipe that was continually re-evaluated even though the underlying values had not changed.

The JS profile tab showed where the problem was.

**Comment by tbosch** [comment]
The reason is that `enableProdMode` does not really work any more.
Investigating...

**Comment by ericmartinezr** [comment]
@tbosch fwiw https://github.com/angular/angular/issues/8508

**Comment by tbosch** [comment]
Reason: We split our facades for core and compiler. So the `assertionsEnabled()` check which lives in the faces does not reach the compiler!

**Comment by tbosch** [comment]
Workaround: 

1) import and call `enableProdMode` for all of the following modules:

- `@angular/core/src/facade/lang`
- `@angular/compiler/src/facade/lang`
- `@angular/platform-browser/src/facade/lang`

2) Provide a binding for `CompilerConfig` directly that disables debug info (only for RC2, RC1 did not have this problem):
```
import {CompilerConfig} from '@angular/compiler';

bootstrap(MyComponent, [{
  provide: CompilerConfig, useValue: new CompilerConfig({genDebugInfo: false, logBindingUpdate: false})
}]);
```

**Comment by vicb** [comment]
closed via #8508
